### PR TITLE
Local volumes for local storage operator

### PR DESCRIPTION
**Pull request by adtandale**

### CODE REVIEW

File: piqe_ocp_lib/api/crd/local_storage_operator.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        self.api_version = "local.storage.openshift.io/v1"
+        self.kind = "LocalVolume"
+        self.lv = self.dyn_client.resources.get(api_version=self.api_version, kind=self.kind)
+
+    def create_local_volume(self) -> Optional[ResourceInstance]:
+        """
+        create local volume
+        :return: api reponse
+        """
+        if self.check_operator_install is not None and self.version is not None and self.channel is not None:
```
Comment (btjd) [inline piqe_ocp_lib/api/crd/local_storage_operator.py:50]: more nitpick :) you can make this long if statement more 'pythonic' as suggested in this link
https://newbedev.com/what-is-the-most-pythonic-way-to-check-if-multiple-variables-are-not-none

File: piqe_ocp_lib/api/crd/local_storage_operator.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            csv = ClusterServiceVersion()
+            csv_obj = csv.get_cluster_service_version(
+                "local-storage-operator." + self.version, "openshift-local-storage"
+            )
+            crd = csv_obj.metadata.annotations["alm-examples"]
+            for i in range(0, len(eval(crd))):
+                if "'kind': 'LocalVolume', 'metadata'" in str(eval(crd)[i]):
+                    target_item = i
+            body = eval(crd)[target_item]
```
Comment (btjd) [inline piqe_ocp_lib/api/crd/local_storage_operator.py:59]: Are there any values should be modifiable in this body? I would guess we don't necessarily want to use it as is?

File: piqe_ocp_lib/tests/crd/test_local_storage_operator.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return LocalStorageOperator(kube_config_file=get_kubeconfig)
+
+    @pytest.fixture(scope="module")
+    def local_volume(get_kubeconfig) -> LocalVolume:
+        return LocalVolume(kube_config_file=get_kubeconfig)
+
+
+class TestLocalStorageOperator:
+    def test_create_local_volume(self, get_test_objects):
+        assert get_test_objects.lv.create_local_volume() is not None
```
Comment (btjd) [inline piqe_ocp_lib/tests/crd/test_local_storage_operator.py:39]: This works, but maybe also check that we get the right 'kind' of the return object?

File: piqe_ocp_lib/tests/crd/test_local_storage_operator.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def test_get_local_volume(self, get_test_objects):
+        api_response = get_test_objects.lv.get_local_volume()
+        assert len(api_response.items) != 0
+
+    def test_delete_local_volume(self, get_test_objects):
+        assert get_test_objects.lv.delete_local_volume() is not None
+
+    def test_watch_local_volume(self, get_test_objects):
+        assert get_test_objects.lv.watch_local_volume("example") is not False
```
Comment (btjd) [inline piqe_ocp_lib/tests/crd/test_local_storage_operator.py:49]: When does the 'example' local volume get created? would it make more sense to run this test between the create and delete tests?
Also, would this work on any vanilla cluster or do you need a specific setup?

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Comment by adtandale** [comment]
retest

**Review by btjd — APPROVED** [review]
LGTM, please make sure to clean up comments and squash. Thank you

**Review by Dannyb48 — APPROVED** [review]
LGTM

**Pull request merged**